xds/internal/xdsclient/xds.go

```diff
@@ -561,6 +568,40 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger,
 	return routesRet, nil
 }
 
+func hashPoliciesProtoToSlice(policies []*v3routepb.RouteAction_HashPolicy) ([]*HashPolicy, error) {
+	var hashPoliciesRet []*HashPolicy
+	for _, p := range policies {
+		// Hash Policies are only applicable for a Ring Hash LB.
+		if !env.RingHashSupport {
```

Comment:
Do this check before calling `hashPoliciesProtoToSlice`.

And also, no need to return an error (error results in NACKs). It's safe to just ignore it.

Reply:
Wrapped the hashPoliciesProtoToSlice call inside a RingHashSupport conditional.